pilopt/src/lib.rs

```diff
@@ -558,6 +563,178 @@ fn extract_constant_lookups<T: FieldElement>(pil_file: &mut Analyzed<T>) {
     }
 }
 
+/// Identifies witness columns that are constrained to a non-shifted (multi)linear expression, replaces the witness column by an intermediate polynomial
```

Comment:
Can you extend the comment a little? What is the intermediate polynomial and why are we doing this optimization?